test/computDYsplitting_PT.cpp

```diff
@@ -0,0 +1,211 @@
+// open input txt file and append all the files it contains to TChain
+void appendFromFileList (TChain* chain, TString filename)
+{
+  std::ifstream infile(filename.Data());
+  std::string line;
+  while (std::getline(infile, line))
+  {
+    line = line.substr(0, line.find("#", 0)); // remove comments introduced by #
+    while (line.find(" ") != std::string::npos) line = line.erase(line.find(" "), 1); // remove white spaces
+    while (line.find("\n") != std::string::npos) line = line.erase(line.find("\n"), 1); // remove new line characters
+    while (line.find("\r") != std::string::npos) line = line.erase(line.find("\r"), 1); // remove carriage return characters
+    if (!line.empty()) // skip empty lines
+      chain->Add(line.c_str());
+  }
+  return;
+}
+
+int main(int argc, char** argv)
+{
+  if (argc < 2)
+  {
+    cout << "** error of inputs . Usage:" << endl;
+    cout << "computeDYsplitting <inputFileList> <OPTIONAL : outputFileName>" << endl;
+    return 1;
+  }
+
+  string inputFile = argv[1];
+  string outputFileName = "histos_DYCount.root" ;
+  if (argc > 2) outputFileName = argv[2];
+
+  cout << "** INFO: using filelist : " << inputFile << endl;
+
+  // input and output setup
+  // ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+  TChain * bigChain = new TChain ("HTauTauTree/HTauTauTree") ;
+  appendFromFileList (bigChain, inputFile);
+  bigTree theBigTree (bigChain) ;
+
+  int totEvents = 0;
+  int allEvts[5][5][6]; // njet,nb,ht
+  for (int i = 0; i < 5; i++)
+    for (int j = 0; j < 5; j++)
+      for (int k = 0; k < 6; k++)
+	allEvts[i][j][k] = 0;
+
+  float xbins[6]  = {0.0, 1.0, 2.0, 3.0, 4.0, 5.0};
+  float ybins[6]  = {0.0, 1.0, 2.0, 3.0, 4.0, 5.0};
+  float zbins[7]  = {0.0, 50., 100., 250., 400., 650., 10000.};
+
+  TFile* fOut = new TFile (outputFileName.c_str(), "recreate");
+  TH3F* h_nJets_nBs_pt    = new TH3F ("h_nJets_nBs_pt"   , ";nJets;nBs;pt", 5, xbins, 5, ybins, 6, zbins);
+  TH3I* hINT_nJets_nBs_pt = new TH3I ("hINT_nJets_nBs_pt", ";nJets;nBs;pt", 5, xbins, 5, ybins, 6, zbins);
+
+  // speed up
+  theBigTree.fChain->SetBranchStatus("*", 0);
+  theBigTree.fChain->SetBranchStatus("lheNOutPartons", 1);
+  theBigTree.fChain->SetBranchStatus("lheNOutB", 1);
+  //theBigTree.fChain->SetBranchStatus("lheHt", 1);
+  theBigTree.fChain->SetBranchStatus("genpart_px",1);
+  theBigTree.fChain->SetBranchStatus("genpart_py",1);
+  theBigTree.fChain->SetBranchStatus("genpart_pz",1);
+  theBigTree.fChain->SetBranchStatus("genpart_e",1);
+  theBigTree.fChain->SetBranchStatus("genpart_flags",1);
+  theBigTree.fChain->SetBranchStatus("genpart_pdg",1);
+
+
+  cout << "** INFO: most branches are desactivated to speed up reading..." << endl;
+  // loop over events
+  // ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+  for (Long64_t iEvent = 0 ; true ; ++iEvent)
+  {
+    if (iEvent % 10000 == 0)  cout << "- reading event " << iEvent << endl ;
```

Comment:
Very minor: i would suggest to increase this to 100000 in order to not flood the terminal output.